pages/runtime/data-api.md

```diff
@@ -0,0 +1,672 @@
+------|------|------------:
+ID | `INTEGER` | device ID
+
+* cURL Example:
+
+```
+curl "https://api.resin.io/v1/device(<ID>)?\$select=ip_address" 
+-H "Content-Type: application/json"
+-H "Authorization: Bearer <auth_token>"
+```
+
+#### Check if web accessible
+* Summary: Check if the specified device is web accessible
+* Endpoint: `/device(<ID>)?$select=is_web_accessible`
+* Method: `GET`
+
+Param | Type | Description
+------|------|------------:
+ID | `INTEGER` | device ID
+
+* cURL Example:
+
+```
+curl "https://api.resin.io/v1/device(<ID>)?\$select=is_web_accessible" 
+-H "Content-Type: application/json"
+-H "Authorization: Bearer <auth_token>"
+```
+
+#### Identify device
+* Summary: Identify device (blink) given the device UUID
+* Endpoint: `https://api.resin.io/blink`
+* Method: `POST`
+
+Param | Type | Description
+------|------|------------:
+uuid | `STRING` | device UUID
+
+* cURL Example:
+
+```
+curl -X POST "https://api.resin.io/blink" 
+-H "Content-Type: application/json" 
+-H "Authorization: Bearer <auth_token>"
+--data '{"uuid": "<uuid>"}'
+```
+
+#### Rename device
+* Summary: Rename the device with the specified ID, with the given new name
+* Endpoint: `/device(<ID>)`
+* Method: `PATCH`
+
+Param | Type | Description
+------|------|------------:
+ID | `INTEGER` | device ID
+name | `STRING` | device new name
+
+* cURL Example:
+
+```
+curl -X PATCH "https://api.resin.io/v1/device(<ID>)" 
+-H "Content-Type: application/json" 
+-H "Authorization: Bearer <auth_token>"
+--data '{"name": "<new_name>"}'
+```
+
+#### Delete device
+* Summary: Delete a specific device
+* Endpoint: `/device(<ID>)`
+* Method: `DELETE`
+
+Param | Type | Description
+------|------|------------:
+ID | `INTEGER` | device ID
+
+* cURL Example:
+
+```
+curl -X DELETE 'https://api.resin.io/v1/device(<ID>)' 
+-H 'Content-Type: application/json' 
+-H 'Authorization: Bearer <auth_token>'
+```
+
+#### Add note
+* Summary: Add note to a specific device
+* Endpoint: `/device(<ID>)`
+* Method: `PATCH`
+
+Param | Type | Description
+------|------|------------:
+ID | `INTEGER` | device ID
+note | `STRING` | device new note
+
+* cURL Example:
+
+```
+curl -X PATCH "https://api.resin.io/v1/device(<ID>)" 
+-H "Content-Type: application/json" 
+-H "Authorization: Bearer <auth_token>" 
+--data '{"note": "<new note>"}'
+```
+
+#### Restart application
+* Summary: Restart application container on the specified device
+* Endpoint: `https://api.resin.io/device/<ID>/restart`
+* Method: `POST`
+
+Param | Type | Description
+------|------|------------:
+ID | `INTEGER` | device ID
+
+* cURL Example:
+
+```
+curl -X POST "https://api.resin.io/device/<ID>/restart"
+-H "Content-Type: application/json" 
+-H "Authorization: Bearer <auth_token>"
+```
+
+#### Reboot
+* Summary: Reboot the device with the specified UUID
+* Endpoint: `https://api.resin.io/supervisor/v1/reboot`
+* Method: `POST`
+
+Param | Type | Description
+------|------|------------:
+uuid | `STRING` | device UUID
```

Comment:
This can alternatively be `device: $deviceID` or `application: $applicationID` (which allows rebooting all devices of a given application at once), or any combination of the 3.

More generally the endpoint is actually `https://api.resin.io/supervisor/$supervisorApiEndpoint` with uuid/application/device/method/data parameters, with uuid/application/device being used to specify the device for which to make the supervisor api call and `method` being the method to use (PUT/PATCH/POST/HEAD/DELETE/GET), default to `POST`, and `data` being a JSON object that will be forwarded to the supervisor api (default to null/empty)

So for these endpoints I think it makes more sense to reference the supervisor api as their functionality is governed by the supervisor on the device